app/processing/delinked-statement/get-delinked-statement-by-payment-reference.js

```diff
@@ -0,0 +1,73 @@
+const getD365 = require('./d365')
+const getOrganisation = require('./organisation')
+const getDelinkedCalculation = require('./delinkedCalculation')
+const saveDocument = require('./save-document')
+const getPreviousPaymentCountByCalculationId = require('./get-previous-payment-count-by-calculation-id')
+const getDocumentTypeByCode = require('./get-document-type-by-code')
+const getAddressFromOrganisation = require('./get-address-from-organisation')
+const { DELINKED } = require('../../constants/document-types')
+
+const getDelinkedStatementByPaymentReference = async (paymentReference, _excluded) => {
+  const delinkedMarketingYear = '2024'
+  const delinkedFullName = 'Delinked Payment Statement'
+  const delinkedShortName = 'DP'
+
+  const d365 = await getD365(paymentReference)
+  if (!d365) {
+    throw new Error('D365 data not found')
+  }
+
+  const delinkedCalculation = await getDelinkedCalculation(d365.calculationId)
+  if (!delinkedCalculation) {
+    throw new Error('Delinked calculation data not found')
```

Comment:
Very generic log, same as above. Should specify for which key element data was not found. In this case d65.calculationId